examples/src/org/ejml/example/ExampleSemiRings.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2009-2020, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml.example;
+
+import org.ejml.data.DMatrixSparseCSC;
+import org.ejml.ops.DMonoids;
+import org.ejml.ops.DSemiRing;
+import org.ejml.ops.DSemiRings;
+import org.ejml.sparse.csc.CommonOpsWithSemiRing_DSCC;
+
+/**
+ * Example showing the usage of semirings in the context of graphs as for instance used in GraphBLAS
```

Comment:
Could you explain very briefly which problem you are solving in the description. Also might want to change the class name if possible. E.g.
`Example showing how to find the lowest cost path in a graph using semirings`

Then `ExampleGraphBestPath`

More people will understand that language and might realize this is useful for them.